flow/Arena.h

```diff
@@ -965,9 +962,9 @@ class VectorRef : public ComposedIdentifier<T, 0x8>, public VectorRefPreserializ
 	void reallocate(Arena& p, int requiredCapacity) {
 		requiredCapacity = std::max(m_capacity * 2, requiredCapacity);
 		// SOMEDAY: Maybe we are right at the end of the arena and can expand cheaply
-		T* newData = (T*)new (p) uint8_t[requiredCapacity * sizeof(T)];
+		T* newData = new (p) T[requiredCapacity];
```

Comment:
I didn't see a noticeable performance regression from this change, but it could happen in theory because we're now calling the default constructor many times. It looks like we can get away with using memcpy here if necessary.